add test for input overflow text preserved

boazsender opened the pull request:
This is a tentative test for the treatment of text in a text input where the padding and boarder are greater than the height of the test.

~~This should land if the discussion in w3c/csswg-drafts/issues/1717 arrives at specifying the behavior that chrome, edge and safari agree on and with which firefox does not: to preserve text height in an input with padding and border greater than the height of the text.~~

~~While the gecko behavior is technically more correct, there are enough compat issues caused by this behavior referenced on whatwg/compat/issues/81 that I believe the behavior in this ref test will be specified.~~

**Update**: w3c/csswg-drafts/issues/1717 has been closed in favor of w3c/csswg-drafts/issues/1941, and the firefox project have decided to implement the behavior described by this test. Here is their bug for this work: https://bugzilla.mozilla.org/show_bug.cgi?id=752790

~~There should be an accompanying spec change to this test (likely in css-overflow, but potentially in css-align) which I'm still looking for help on in w3c/csswg-drafts/issues/1717.~~

**Update**: I've asked for guidance on where to make the spec change on w3c/csswg-drafts/issues/1941.

cc @alijuma and @Hexcles who I originally wrote this reference with. And thank you @frivoal for pointing me toward css-overflow.

<!-- Reviewable:start -->

<!-- Reviewable:end -->

Commit:
add test for input overflow text preserved per discussion in https://github.com/w3c/csswg-drafts/issues/1717

w3c-bots commented:
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/20210)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/298139084)

gsnedders commented:
@boazsender Note that since the big change a few days ago these tests need to be in `css/css-overflow`.

Commit:
move input text preservation from css/css-overflow-3 to css/css-overflow per feedback from @gsnedders

Commit:
fix lint error in input scrollable test

Commit:
rename input text preservation tests

Commit:
remove errant css/overflow directory

Commit:
switch to using an input for the reference

boazsender commented:
@gsnedders I've moved the test to css/css-overflow/ and also just updated the reference for this to use an input instead of trying to moch it up with a div. This now consistently passes in chrome and fails in firefox.

Hexcles commented:
Also thumbs up to using Ahem! No more off-by-a-few-pixels. :)

gsnedders commented on css/css-overflow/input-scrollable-region-001.html:
```diff
@@ -0,0 +1,26 @@
+<link rel="author" title="Boaz Sender, Ali Juma, Robert Ma">
+<link rel="help" href="https://drafts.csswg.org/css-overflow-3/#scrollable">
+<link rel="match" href="reference/input-scrollable-region-001-ref.html">
+<meta name="assert" content="input text is preserved when the (input heigh
```
`s/heigh/height/`

gsnedders reviewed (approved):
plz do the spelling fix, but otherwise LGTM

Commit:
fixes typo in test title

boazsender commented:
Typo fixed @gsnedders. Thanks for the review!
